src/main/java/seedu/duke/storage/FoodListDecoder.java

```diff
@@ -0,0 +1,34 @@
+package seedu.duke.storage;
+
+import seedu.duke.food.Food;
+import seedu.duke.food.FoodList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Decodes the storage data file into an {@code FoodList} object.
+ * The code is adapted from
+ * https://github.com/se-edu/addressbook-level2/blob/master/src/seedu/addressbook/storage/AddressBookDecoder.java
+ */
+public class FoodListDecoder {
+
+    /**
+     * Decodes {@code encodedFoodList} into an {@code FoodList} containing the decoded food.
+     */
+    public static FoodList decodeFoodList(List<String> encodedFoodList) {
+        final ArrayList<Food> decodedFoodList = new ArrayList<>();
+        for (String encodedFood : encodedFoodList) {
+            decodedFoodList.add(decodeFoodFromString(encodedFood));
+        }
+        return new FoodList(decodedFoodList);
+    }
+
+    /**
+     * Decodes {@code encodedFood} into a {@code Food}.
+     */
+    private static Food decodeFoodFromString(String encodedFood) {
+        String[] details = encodedFood.split("-e");

```

Comment:
Maybe consider splitting by "-" rather than with "-e" - after having more flags later (quantity, categories, etc.), it will be much easier to scale this method. 

You can then just operate on "e [details]", "q [details]" chunks through splitting flag name from the details

Reply:
The way the details are encoded in the foodListEncoder class will result in the order being the same every time. I feel that operating on flags may be unnecessary.